### DISCUSSION TITLE
Delayed launch not working in beta 72?

### DISCUSSION THREAD

**Discussion by geofftaylor** [ttscoff/bunch-docs#32] [Q&A] [answered]
[Delayed launch](https://brettterpstra.com/bunch-beta/docs/bunch-files/delay/) doesn't seem to work in beta 72.

I created a bunch that launches an app and then arranges windows using Moom (via AppleScript). It doesn't work as expected because the app takes a couple of seconds to draw its window, and the AppleScript line executes before the app has fully launched.  (That's what I suspect is happening anyway.)

My quick and dirty solution was to add a delay to the AppleScript line, but even then it was executing as soon as I ran the bunch.

I created another bunch to test the delay:

```
TextEdit ~5
* display dialog "Was the launch delayed?" ~10
```

When I run this bunch, TextEdit and the AppleScript dialog both appear immediately. Am I missing something with the syntax, or am I using the feature incorrectly?

**Comment by ttscoff** [comment]
Delays are working for me, but they only work with serial sequence launches. If you've enabled parallel execution in Preferences, you can either disable that, or just add `sequence: serial` to your Bunch's frontmatter. Let me know if that fixes delays for you.

The idea behind the Waiting Snippet is cool, but the fact is that macOS doesn't receive notifications after launch, which apps can send long before they've actually displayed their windows. So the Waiting Snippet holds off until everything has sent its didFinishLaunching notification, but in a lot of cases that's still going to require a delay. I'll try to fix delays on parallel execution, at least for Waiting Snippets.

**Comment by ttscoff** [comment, marked as answer]
Ok, I have this fixed up for the b74 update, which should be out today/tomorrow. It should now allow delays on any item, including Waiting Snippets, even in parallel execution.

I'll need you to test a bit once you get the update, so please check back in and let me know if you still have any issues.

**Reply by ttscoff** [reply]
Ok, b74 is out, please let me know how it goes!

**Reply by geofftaylor** [reply]
It's working now, even with parallel execution enabled. Thanks!
